tests/Unit/Evolution/Systems/NewtonianEuler/Test_Sources.cpp

```diff
@@ -0,0 +1,164 @@
+
+  using argument_tags =
+      tmpl::list<FirstArg, SecondArg<Dim>, ThirdArg, FourthArg<Dim>>;
+
+  void apply(
+      const gsl::not_null<tnsr::I<DataVector, Dim>*> source_momentum_density,
+      const gsl::not_null<Scalar<DataVector>*> source_energy_density,
+      const Scalar<DataVector>& first_arg,
+      const tnsr::I<DataVector, Dim>& second_arg,
+      const Scalar<DataVector>& third_arg,
+      const tnsr::i<DataVector, Dim>& fourth_arg) const noexcept {
+    for (size_t i = 0; i < Dim; ++i) {
+      source_momentum_density->get(i) =
+          (get(first_arg) - 1.5 * get(third_arg)) * second_arg.get(i);
+    }
+    get(*source_energy_density) =
+        get(dot_product(second_arg, fourth_arg)) + 3.0 * get(third_arg);
+  }
+};
+
+// We need this proxy, as pypp can't work with arguments other than doubles
+// and SpECTRE tensors.
+template <typename SourceTermType>
+struct ComputeSourcesProxy {
+  static constexpr size_t dim = SourceTermType::volume_dim;
+  static void apply(
+      const gsl::not_null<Scalar<DataVector>*> source_mass_density_cons,
+      const gsl::not_null<tnsr::I<DataVector, dim>*> source_momentum_density,
+      const gsl::not_null<Scalar<DataVector>*> source_energy_density,
+      const Scalar<DataVector>& first_arg,
+      const tnsr::I<DataVector, dim>& second_arg,
+      const Scalar<DataVector>& third_arg,
+      const tnsr::i<DataVector, dim>& fourth_arg) noexcept {
+    SourceTermType source_computer;
+    NewtonianEuler::ComputeSources<SourceTermType>::apply(
+        source_mass_density_cons, source_momentum_density,
+        source_energy_density, source_computer, first_arg, second_arg,
+        third_arg, fourth_arg);
+  }
+};
+
+// Pypp needs explicit args for it to work, so we need a new proxy for testing
+// second source term with different number of sourced variables.
+template <typename SourceTermType>
+struct ComputeSourcesProxy2 {
+  static constexpr size_t dim = SourceTermType::volume_dim;
+  static void apply(
+      const gsl::not_null<tnsr::I<DataVector, dim>*> source_momentum_density,
+      const gsl::not_null<Scalar<DataVector>*> source_energy_density,
+      const Scalar<DataVector>& first_arg,
+      const tnsr::I<DataVector, dim>& second_arg,
+      const Scalar<DataVector>& third_arg,
+      const tnsr::i<DataVector, dim>& fourth_arg) noexcept {
+    SourceTermType source_computer;
+    NewtonianEuler::ComputeSources<SourceTermType>::apply(
+        source_momentum_density, source_energy_density, source_computer,
+        first_arg, second_arg, third_arg, fourth_arg);
+  }
+};
+
+template <size_t Dim>
+void test_sources(const DataVector& used_for_size) {
+  pypp::check_with_random_values<4>(
+      &ComputeSourcesProxy<SomeSourceType<Dim>>::apply, "TestFunctions",
+      {"source_mass_density_cons", "source_momentum_density",
+       "source_energy_density"},
+      {{{-1.0, 1.0}, {-1.0, 1.0}, {-1.0, 1.0}, {-1.0, 1.0}}}, used_for_size);
```

Comment:
since all the bounds are the same, you can just pass a single `{-1.0, 1.0}` and change the `4` on line 144 to `1`